frontend/workflows/projectSelector/src/selector-reducer.tsx

```diff
@@ -1,9 +1,31 @@
 import type { clutch as IClutch } from "@clutch-sh/api";
 import _ from "lodash";
 
-import type { Action, State } from "./hello-world";
+import type { Action, ProjectState, State } from "./hello-world";
 import { deriveSwitchStatus, Group } from "./hello-world";
 
+const updateGroupstate = (
+  state: State,
+  group: Group,
+  project: string,
+  projectState: ProjectState,
+): State => {
+  const newState = { ...state };
+  if (project in newState[group]) {
+    // preserve the checked value if the project is already in the group
+    newState[group][project].checked = state[group][project].checked;
+    if (group === Group.PROJECTS) {
```

Comment:
this isn't really needed, as we never pass in a custom true on a non group project